src/core/app-template.cc

```diff
@@ -211,7 +223,8 @@ app_template::run_deprecated(int ac, char ** av, std::function<void ()>&& func)
         if (!_opts.description.empty()) {
             std::cout << _opts.description << "\n";
         }
-        std::cout << _app_opts << "\n";
+        auto& which_opts = _app_visible_opts_for_help ? *_app_visible_opts_for_help : _app_opts;
```

Comment:
if we use `&_app_opts` as the default value of `_app_visible_opts_for_help`. we could simplify this part like:
```c++
        assert(_app_visible_opts_for_help);
        std::cout << *_app_visible_opts_for_help << "\n";
```